AK/Utf16View.cpp

```diff
@@ -37,35 +37,40 @@ static constexpr u16 host_code_unit(u16 code_unit, Endianness endianness)
 }
 
 template<OneOf<Utf8View, Utf32View> UtfViewType>
-static ErrorOr<Utf16Data> to_utf16_slow(UtfViewType const& view, Endianness endianness)
+static ErrorOr<Utf16ConversionResult> to_utf16_slow(UtfViewType const& view, Endianness endianness)
 {
     Utf16Data utf16_data;
     TRY(utf16_data.try_ensure_capacity(view.length()));
 
-    for (auto code_point : view)
+    size_t code_point_count = 0;
+    for (auto code_point : view) {
         TRY(code_point_to_utf16(utf16_data, code_point, endianness));
+        code_point_count++;
+    }
 
-    return utf16_data;
+    return Utf16ConversionResult { utf16_data, code_point_count };
```

Comment:
move the data? or construct the result at the top of the function and use result.data, result.count? I'm not convinced you're getting the same NRVO here as before.